Content.Server/EntityEffects/Effects/ActivateRandom.cs

```diff
@@ -0,0 +1,166 @@
+using Content.Shared.FixedPoint;
+using Content.Shared.EntityEffects;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using JetBrains.Annotations;
+using System.Linq;
+
+namespace Content.Shared.EntityEffects.Effects;
+
+/// <summary>
+/// "Activates" random of predefined sets of chemicals, removing them from the solution
+/// and firing specified ReagentEffects once.
+/// </summary>
+[UsedImplicitly]
+public sealed partial class ActivateRandom : EntityEffect {
+    /// <summary>
+    /// List of sets of activatable chemicals along with results of the activation.
+    /// </summary>
+    [DataField(required: true)]
+    public ActivateRandomCase[] Cases;
+
+    protected override string? ReagentEffectGuidebookText(IPrototypeManager prototype, IEntitySystemManager entSys)
+    {
+        return Loc.GetString("reagent-effect-guidebook-activate-random",
+            ("chance", Probability),
+            ("cases", string.Join("\n", Cases.ToList().Select(case_ =>
+                "- "
+                + string.Join(", ", case_.Activates.ToList()
+                    .Select(activate =>
+                        prototype.TryIndex(activate.Reagent.Prototype, out ReagentPrototype? reagentProto)
+                            ? Loc.GetString("reagent-effect-guidebook-activate-random-activate",
+                                ("factor", activate.Factor),
+                                ("catalyst", !activate.Consume),
+                                ("reagent", reagentProto.LocalizedName))
+                            : null)
+                        .Where(x => x is not null))
+                + ":\n"
+                + string.Join("\n", case_.Effects.ToList()
+                    .Select(effect =>
+                    {
+                        var desc = effect.GuidebookEffectDescription(prototype, entSys);
+                        return desc is null ? null : "  " + desc;
+                    })
+                    .Where(x => x is not null))))
+                + "\n"));
+    }
+
+    public override void Effect(EntityEffectBaseArgs baseArgs)
+    {
+         // ActivateRandom works by activating other reagents, it needs access to extended information
+        if (baseArgs is not EntityEffectReagentArgs)
+            return;
+        EntityEffectReagentArgs args = (EntityEffectReagentArgs) baseArgs;
+            
+        if (args.Source is null)
+            return; // ActivateRandom activates chemicals from the args.Source.
+            // If source is not present, ActivateRandom has nothing to activate.
+
+        // Only one random set of chemicals is activated. The chance to activate
+        // some speicific set is based on how many of these chemicals (with respect to factor)
+        // are present in the solution.
+        // We use chance and chanceSum to calculate proprotions of chemical sets (with respect to factor)
+        // present in the solution.
+        float[] chances = new float[Cases.Length];
```

Comment:
Same here we use var for styleguide.
```suggestion
        // present in the solution.
        var chances = new float[Cases.Length];
```